### PR TITLE
Unauthorized Page 

### PR DESCRIPTION
**Pull request by jdtoombs**
### Motivation and Context
The application should display an informative message if the user does not have access to it. 

### Description
If the user logs in without the appropriate group/role they will be presented with a message to contact their admin, and that they do not have the appropriate roles. The fronted will wait for the back end to fetch the user info before determining to display this page or not.
image

Additionally, a new group has been added to Entra ID called q-pilot, which is now required to access the application. I have added all of us as "owners".
image

I have also added a script that will run when you `./scripts/start-frontend.sh` to add the default guid for the appropriate group unless otherwise specified in your `/webapp/.env`.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: webapp/env.sh [hunk 1]
```diff
@@ -24,10 +24,30 @@ do
...
   
   # Append configuration property to JS file
   echo "  $varname: \"$value\"," >> ./env-config.js
+
 done < .env
 
-echo "}" >> ./env-config.js
\ No newline at end of file
+# Add a default security group ID if it's not in .env
+security_group_default="e1aae29c-d680-4e7e-a90a-d10cdb18fd74"
```
Comment (jdtoombs) [inline webapp/env.sh:35]: Wondering if there is a more friendly way to identify the group/role, I viewed the active user info and pulled this guid from my groups claim. Also wondering if these are the same across environments. You do have the ability to overwrite in your .env file

Comment (wdavidcamden) [inline webapp/env.sh:35]: Not all users are in the Admin Group and each specialization could have a different set of Groups. I guess we could look at a base group that is used for the just to determine if they have access to the app and other groups for the specialization.

File: webapp/env.sh [hunk 2]
```diff
@@ -24,10 +24,30 @@ do
...
+security_group=$(grep -E '^SECURITY_GROUP_ID=' .env | cut -d '=' -f2)
+
+# If SECURITY_GROUP_ID not found in .env, use the default value
+if [[ -z "$security_group" ]]; then
+  echo "SECURITY_GROUP_ID not found in .env, using default."
```
Comment (wdavidcamden) [inline webapp/env.sh:40]: I would put the default security group into the .env file, and allow the scripts environment variable check override the group at that point

File: webapp/env.sh [hunk 3]
```diff
@@ -24,10 +24,30 @@ do
   # Read value of current variable if exists as Environment variable
   value=$(printf '%s\n' "${!varname}")
   # Otherwise use value from .env file
-  [[ -z $value ]] && value=${varvalue}
+  [[ -z "$value" ]] && value="$varvalue"
```
Comment (wdavidcamden) [inline webapp/env.sh:27]: was this changed for a reason? this worked before...

File: webapp/env.sh (hunk 1 above)
Comment (wdavidcamden) [reply webapp/env.sh:35]: sadly EntraID only provides a GUID and not a friendly name.

File: webapp/env.sh (hunk 3 above)
Comment (jdtoombs) [reply webapp/env.sh:27]: i can remove it, was just to satisfy my linter for some reason :stuck_out_tongue_closed_eyes:

File: webapp/env.sh (hunk 1 above)
Comment (jdtoombs) [reply webapp/env.sh:35]: Not a bad idea! In my last project we did something similar

Comment (jdtoombs) [reply webapp/env.sh:35]: @wdavidcamden should I make a new "Base Group" or is there an existing group I could use?

Comment (wdavidcamden) [reply webapp/env.sh:35]: we will need a new one in Azure EntraID. maybe just q-pilot? oh and we might as well rename the other groups from copilot_something to q-pilot_something

**Comment by wdavidcamden** [comment]
> @wdavidcamden - I've update the PR - my one concern is locking out existing users, I believe I have everyone added.. also gave us all permissions to add people to this group. Maybe if you can verify for me [here to be safe](https://portal.azure.com/#view/Microsoft_AAD_IAM/GroupDetailsMenuBlade/~/Overview/groupId/4e802d2e-65dc-4bb8-98bc-b08a7d0ed53d/menuId/) :)

We actually might have too many users now :) . maybe if we just include the other existing groups as members of the base one for now, that should work.

**Review by wdavidcamden — APPROVED** [review]

**Pull request merged**